### PR TITLE
refactor(strm-2696): some cleanup and various minor improvements

### PR DESCRIPTION
**Pull request by ivan-p92**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -32,33 +32,34 @@ plugins {
...
+
+    // Self-managed dependencies
+    implementation("net.devh:grpc-server-spring-boot-starter:2.15.0.RELEASE")
     implementation("com.databricks:databricks-sdk-java:0.10.0")
 
     implementation("com.nimbusds:nimbus-jose-jwt:9.37")
     implementation("org.bouncycastle:bcpkix-jdk18on:1.76")
 
-    // Todo remove before squashing
+    // Todo remove before squashing - why?
```
Comment (trietsch) [inline app/build.gradle.kts:55]: Does not have to be removed IMO.

File: app/src/main/kotlin/com/getstrm/pace/api/DataPolicyApi.kt [hunk 2]
```diff
@@ -31,6 +31,7 @@ import com.getstrm.pace.service.DataPolicyService
 import com.getstrm.pace.service.ProcessingPlatformsService
 import net.devh.boot.grpc.server.service.GrpcService
 
+// Todo: probably split in two separate services, depending on upcoming proto definition changes
```
Comment (trietsch) [inline app/src/main/kotlin/com/getstrm/pace/api/DataPolicyApi.kt:34]: We'll do this on the branch where we've changed the proto files.

File: app/src/main/kotlin/com/getstrm/pace/bigquery/BigQueryClient.kt [hunk 3]
```diff
@@ -66,7 +73,7 @@ class BigQueryClient(
                 InternalException.Code.INTERNAL,
                 DebugInfo.newBuilder()
                     .setDetail(
-                        "Error while executing BigQuery query (error message: ${e.message}), please check the logs of your PACE deployment. This is a bug, please report to https://github.com/getstrm/pace/issues/new"
+                        "Error while executing BigQuery query (error message: ${e.message}), please check the logs of your PACE deployment. $BUG_REPORT"
```
Comment (trietsch) [inline app/src/main/kotlin/com/getstrm/pace/bigquery/BigQueryClient.kt:76]: 👌

File: app/src/main/kotlin/com/getstrm/pace/catalogs/OpenDataDiscovery.kt [hunk 4]
```diff
@@ -94,7 +99,7 @@ class OpenDataDiscoveryCatalog(configuration: CatalogConfiguration) : DataCatalo
         DataCatalog.Table(schema, id, name) {
 
         private fun getAllParents(parentId: Long, fieldsById: Map<Long, DataSetField>): List<Long> {
-            val parent = fieldsById[parentId]!!
+            val parent = checkNotNull(fieldsById[parentId]) { "The parent field should exist" }
```
Comment (trietsch) [inline app/src/main/kotlin/com/getstrm/pace/catalogs/OpenDataDiscovery.kt:102]: Are not null checks caught somewhere? Or are they propagated to the end user as an `InternalException`?

File: app/src/test/kotlin/com/getstrm/pace/service/RuleSetServiceTest.kt [hunk 5]
```diff
@@ -122,7 +122,7 @@ class RuleSetServiceTest {
 
         runBlocking {
             val policyWithRulesets = underTest.addRuleSet(dataPolicy)!!
-//            println(policyWithRulesets.toYaml().strip())
+//            println(policyWithRulesets.com.getstrm.pace.util.toYaml().strip())
```
Comment (trietsch) [inline app/src/test/kotlin/com/getstrm/pace/service/RuleSetServiceTest.kt:125]: find and replace error?

**Review by trietsch — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/getstrm/pace/catalogs/OpenDataDiscovery.kt (hunk 4 above)
Comment (ivan-p92) [reply app/src/main/kotlin/com/getstrm/pace/catalogs/OpenDataDiscovery.kt:102]: `checkNotNull` throws `IllegalStateExceptions`, while `requireNotNull` throw `IllegalArgumentExceptions`. Both do smart casts. The `!!` would have thrown an NPE. Arguably this isn't much better. Perhaps we should use an elvis with a `PaceStatusException`? Although an internal exception is perhaps fine in this exceptional case.

File: app/src/test/kotlin/com/getstrm/pace/service/RuleSetServiceTest.kt (hunk 5 above)
Comment (ivan-p92) [reply app/src/test/kotlin/com/getstrm/pace/service/RuleSetServiceTest.kt:125]: Sort of, this was because the util file had no package yet.

**Pull request merged**